mitxgraders/comparers/comparers.py

```diff
@@ -112,30 +113,108 @@ class EqualityComparer(Comparer):
+        expected_eval = comparer_params_eval[0]
+        self.validate(expected_eval, student_eval, utils)
+
+        expected_eval = self.transform(expected_eval)
+        student_eval = self.transform(student_eval)
+
+        return utils.within_tolerance(expected_eval, student_eval)
 
 equality_comparer = EqualityComparer()
 
+class MatrixEntryComparer(EqualityComparer):
+    """
+    Default comparer for MatrixGrader. Compares student and instructor matirx
+    evaluations entry-by-entry for equality.
+
+    Configuration
+    =============
+
+        transform (None | function): same as EqualityComparer, defaults to None
+        entry_partial_credit ('proportional'|number): Determines how partial credit
+            is awarded. If set to 'proportional', then credit is proportional to
+            the number of correct matrix entries. If a numeric value is provided,
+            this flat rate of partial credit is provided as long as some but
+            not all entries are correct.
+
+            Default is the numeric value 0 (no partial credit).
+        entry_partial_msg (str): A text string message shown when partial credit
+            is awarded. The string may optionally contain the formatting key {error_indices},
+            which will be replaced with the indices of the incorrect matrix entries.
+    """
+
+    default_msg = "Matrix entries at indices {error_indices} are incorrect."
+    schema_config = EqualityComparer.schema_config.extend({
+        Required('entry_partial_credit', default=0): Any(All(Number, Range(0, 1)), 'proportional'),
+        Required('entry_partial_msg', default=default_msg): text_string
+    })
+
+    @staticmethod
+    def format_message_with_locations(format_string, locs):
+        """
+        Returns format_string with {error_indices} replaced by 1-indexed error
+        locations.
+
+        Arguments:
+            format_string: a string that may contain {error_indices} formatting key.
+            locs: a 0-indexed numpy array of error locations
+        """
+        locs_as_str = ", ".join(map(six.text_type, locs+1))
+        return format_string.format(error_indices=locs_as_str)
+
+    def __call__(self, comparer_params_eval, student_eval, utils):
+        expected_eval = comparer_params_eval[0]
+        self.validate(expected_eval, student_eval, utils)
+
+        num_entries = expected_eval.size
+
+        expected_eval = self.transform(expected_eval)
+        student_eval = self.transform(student_eval)
+
+        vec_within_tol = np.vectorize(utils.within_tolerance)
+        # compared_items is a boolean array of entry-by-entry comparisons
+        compared_items = vec_within_tol(expected_eval, student_eval)
+        percent_correct = np.sum(compared_items).item()/num_entries
+        wrong_locs = np.argwhere(np.logical_not(compared_items))
+        msg = self.format_message_with_locations(self.config['entry_partial_msg'], wrong_locs)
```

Comment:
To account for `entry_partial_msg is None`, just put in an if statement here that otherwise sets `msg = ''`.